src/InstrumentationEngine/CorProfilerInfoWrapper.cpp

```diff
@@ -346,14 +346,29 @@ HRESULT MicrosoftInstrumentationEngine::CCorProfilerInfoWrapper::GetILFunctionBo
     )
 {
     HRESULT hr = S_OK;
+
+    // IDL [out] parameter annotation is optional unless the function only has one.
+    // The corresponding [in] parameter for the size to fill the [out] parameter must be 0.
+    BYTE* pIgnoreMethodHeader;
+    if (ppMethodHeader == NULL)
+    {
+        ppMethodHeader = &pIgnoreMethodHeader;
+    }
+
+    ULONG cbIgnoreMethodSize;
+    if (pcbMethodSize == NULL)
+    {
+        pcbMethodSize = &cbIgnoreMethodSize;
+    }
```

Comment:
Rather than doing this, you could just pass ppMethodHeader and pcbMethodSize into pModuleInfo-GetMethodIl(...). Then you don't have to do any of your own checks. They aren't used in this function except for as out parameters from other functions.

Reply:
The issue is that the CLR allows these out parameters to be optional so I am handling them here by creating dummy local variables that do nothing.

I want these checks to be at the CCorProfilerInfoWrapper level since this function is what is called by the raw profiler.

Reply:
I understand that CLR allows the out parameters to be optional, but that doesn't mean that you have to do any additional work to "ignore them". If they aren't used in the function, then you don't need to have local variables that represent them. Just pass the pointers directly through. You can check them for not-null, and initialize them to avoid having the IfFailRets leaving them as garbage.

The hiccup that you may have with that is that the contract for IMethodInfo and IModuleInfo may be different (do they require the out parameters to be non-null? But, if that is the case, you are going to get inconsistent results from this function regardless.

---
In reply to: [363900865](https://github.com/microsoft/CLRInstrumentationEngine/pull/187#discussion_r363900865) [](ancestors = 363900865)

Reply:
I see what you mean now. I will add null checks to conditionally initialize/set the deref ptr values. 

---
In reply to: [365929212](https://github.com/microsoft/CLRInstrumentationEngine/pull/187#discussion_r365929212) [](ancestors = 365929212,363900865)